posthog/storage/object_storage.py

```diff
@@ -0,0 +1,41 @@
+import structlog
+from boto3 import client
+from botocore.client import Config
+
+logger = structlog.get_logger(__name__)
+
+from posthog.settings import (
+    OBJECT_STORAGE_ACCESS_KEY_ID,
+    OBJECT_STORAGE_BUCKET,
+    OBJECT_STORAGE_HOST,
+    OBJECT_STORAGE_PORT,
+    OBJECT_STORAGE_SECRET_ACCESS_KEY,
+)
+
+s3_client = None
+
+
+# boto doing some magic and gets confused if this is hinted as BaseClient
+# noinspection PyMissingTypeHints
+def storage_client():
+    global s3_client
+    if not s3_client:
+        s3_client = client(
+            "s3",
+            endpoint_url=f"http://{OBJECT_STORAGE_HOST}:{OBJECT_STORAGE_PORT}",
```

Comment:
Nit but I think we should simply provide the whole `endpoint_url` as `{OBJECT_STORAGE_ENDPOINT}`. Hardcoding `http` is already a warning flag to me